api/envoy/config/route/v3/route_components.proto

```diff
@@ -457,6 +457,28 @@ message RouteMatch {
     // Note that CONNECT support is currently considered alpha in Envoy.
     // [#comment:TODO(htuch): Replace the above comment with an alpha tag.
     ConnectMatcher connect_matcher = 12;
+
+    // Supports Google url_template matching for ":path" header.
+    // This match is always case sensitive.
+    // Its syntax is defined at `http.proto <https://github.com/googleapis/googleapis/blob/f829b1334cce86aa3738f3c0698d814b56664445/google/api/http.proto#L208>`_.
+    //
+    // Syntax highlights:
+    //
+    // .. code-block:: text
+    //
+    // * wildcard `*`: matches only 1 path segment.
+    // * double wildcard `**`: matches 0 or more path segments.
+    // * variable `/foo/{x=**}`: group path segments to a variable.
+    // * variable `/foo/{x}`: same as `/foo/{x=*}`
+    //
+    // For examples:
+    //
+    // .. code-block:: text
+    //
+    // * `/foo/*`: /foo/bar should match. But /foo and /foo/bar1/bar2 should not.
```

Comment:
Maybe an example with a suffix as well (common use case), e.g. `/foo/*.js`

Reply:
No,  url_template only works on a whole path segment.  path segments are path split by "/".  it doesn't work on a partial segment. e.g. "*.js"   * is referring to a partial segment.

Reply:
Can you provide some example with a suffix match? I think we do want some expressiveness like this for the cases that @justin-mp was interested in addressing.

Reply:
It doesn't support `/far/*.js` type of suffix match.  It only supports `/**/tail` such suffix match.

Reply:
@justin-mp is this going to cause problems for some of the use cases you were looking at?

Reply:
Yes, this is going to cause problems for our use cases.  We need to be able to match by file name extension.